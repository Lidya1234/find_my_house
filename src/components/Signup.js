import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory, Link } from 'react-router-dom';

import { signuser } from '../reducers/findSlice';
import '../style/style.css';
import '../style/main.scss';
import Notify from '../Notification/Notify';
import Notifyerr from '../Notification/Notifyerr';

const Signup = () => {
  let username = '';
  let password = '';
  /* eslint-disable camelcase */
  let password_confirmation = '';
  const history = useHistory();
  const dispatch = useDispatch();
  const { create } = useSelector((state) => state.houses);
  const handleSubmit = (event) => {
    event.preventDefault();
    const user = {
      username,
      password,
      password_confirmation,
    };

    dispatch(signuser(user));
    setTimeout(() => {
      if (create.status === 500) {
        Notifyerr();
      } else if (create.status === 'created') {
        history.push('/Login');
        Notify();
      }
    }, 6000);
  };

  const handleUNameChange = (e) => {
    username = e.target.value;
  };
  const handlePasswordChange = (e) => {
    password = e.target.value;
  };
  const handlePasswordConfChange = (e) => {
    password_confirmation = e.target.value;
  };
  /* eslint-enable camelcase */
  return (
    <div className="container" data-testid="signup">
      <div className="login">
        <h1>Sign in</h1>
        <span>
          Hello There! Signup and
          <div>start managing your system</div>
        </span>

        <form onSubmit={handleSubmit} className="form">
          <div>
            <input
              type="text"
              placeholder="Enter username"
              className="txt-input"
              name="username"
              onChange={handleUNameChange}
            />
          </div>
          <div>
            <input
              type="password"
              placeholder="Enter password"
              className="txt-input"
              name="password"
              onChange={handlePasswordChange}
            />
          </div>
          <div>
            <input
              type="password"
              placeholder="Confirm password"
              className="txt-input"
              name="password"
              onChange={handlePasswordConfChange}
            />
          </div>
          <div>
            <button type="submit" className="txt-input btn-signin">
              Sign Up
            </button>
          </div>
        </form>
        <div>
          You already have an account
          <Link to="/Login" className="signlink">Sign in</Link>
        </div>
      </div>
    </div>
  );
};
export default Signup;
