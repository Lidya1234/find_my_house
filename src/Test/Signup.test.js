import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../reducers/index';
import Signup from '../components/Signup';

afterEach(() => {
  cleanup();
});

describe('Signup component tests', () => {
  render(
    <Provider store={store}>
      <Router>
        <Signup />
      </Router>
    </Provider>,
  );
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Signup />
        </Router>
      </Provider>,
    )
    .toJSON();

  const signupElement = screen.getByTestId('signup');

  test('Component should Render', () => {
    expect(signupElement).toBeInTheDocument();
  });
  test('Component should contain a sign in test', () => {
    expect(signupElement).toHaveTextContent('Sign in');
  });
  test('Matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
