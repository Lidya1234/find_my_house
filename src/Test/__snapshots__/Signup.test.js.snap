// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component tests Matches snapshot 1`] = `
<div
  className="container"
  data-testid="signup"
>
  <div
    className="login"
  >
    <h1>
      Sign in
    </h1>
    <span>
      Hello There! Signup and
      <div>
        start managing your system
      </div>
    </span>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div>
        <input
          className="txt-input"
          name="username"
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
        />
      </div>
      <div>
        <input
          className="txt-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div>
        <input
          className="txt-input"
          name="password"
          onChange={[Function]}
          placeholder="Confirm password"
          type="password"
        />
      </div>
      <div>
        <button
          className="txt-input btn-signin"
          type="submit"
        >
          Sign Up
        </button>
      </div>
    </form>
    <div>
      You already have an account
      <a
        className="signlink"
        href="/Login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </div>
</div>
`;
